### PR TITLE
Issue 94: Add R and Stan for analytical Weibull

### LINKED ISSUE #94: Implement analytical solution to Weibull with uniform
based on #93 and extending #85 we need to: 

- [ ] Add and test Weibull to R code
- [ ] Add and test Weibull to Stan code
- [ ] Make sure we still have sufficient testing of the numeric solving approaches.

### PR DESCRIPTION
**Pull request by seabbs**
## Description

This PR closes #94 and adds: 

- Translation of the analytical Weibull with uniform solution to R and stan
- Monte Carlo testing of the R analytical solution
- Equivalence testing of the stan and R implementations

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: inst/stan/functions/primary_censored_dist_analytical_cdf.stan [hunk 1]
```diff
@@ -125,6 +125,81 @@ real primary_censored_lognormal_uniform_lcdf(data real d, real q, array[] real p
...
+  *
+  * This function is used in the analytical solution for the primary censored
+  * Weibull distribution with uniform primary censoring. It corresponds to the
+  * g(t; λ, k) function described in the analytic solutions document.
+  *
+  * @param t Upper bound of integration
+  * @param shape Shape parameter (k) of the Weibull distribution
+  * @param scale Scale parameter (λ) of the Weibull distribution
+  *
+  * @return Log of g(t; λ, k) = γ(1 + 1/k, (t/λ)^k) / k - (t/λ) * exp(-(t/λ)^k)
```
Comment (SamuelBrand1) [inline inst/stan/functions/primary_censored_dist_analytical_cdf.stan:139]: This is not quite right

its either 

```math
g(t; \lambda, k) = \gamma(1 + 1/k, (t/\lambda)^k)
```

or

```math
g(t; \lambda, k) = \frac{1}{k}\gamma(1/k, (t/\lambda)^k) - \frac{t}{\lambda}e^{-(t/\lambda)^k}.
```

File: R/primary_censored_dist.R [hunk 2]
```diff
@@ -247,3 +247,84 @@ primary_censored_cdf.pcens_plnorm_dunif <- function(
...
+  inv_shape <- 1 / shape
+  inv_scale <- 1 / scale
+
+  g <- function(t) {
+    # Use the lower incomplete gamma function
+    scaled_t <- (t * inv_scale)^shape
+    gamma_1k <- vapply(scaled_t, function(x) {
+      pracma::gammainc(1 + inv_shape, x)["lowinc"]
+    }, numeric(1)) * inv_shape
+    scale * gamma_1k - t * exp(-scaled_t)
```
[the remark this answers is not in the document]
Comment (seabbs) [reply R/primary_censored_dist.R]: Updated. I think this has gone in between so there is something else wrong

File: inst/stan/functions/primary_censored_dist_analytical_cdf.stan (hunk 1 above)
Comment (seabbs) [reply inst/stan/functions/primary_censored_dist_analytical_cdf.stan:139]: ha yes as above this has got mixed up in the debugging. with either it doesn't work

Comment (seabbs) [reply inst/stan/functions/primary_censored_dist_analytical_cdf.stan:139]: wait

Comment (seabbs) [reply inst/stan/functions/primary_censored_dist_analytical_cdf.stan:139]: I think I havee only every had the 1/k

**Comment by seabbs** [comment]
I think this is all working as expected now. 

I need to do the following: 

- [ ] Open an issue to investigate the analytical solution in R failing when pwindow is large due to the lower incomplete gamma
- [ ] Open an issue to investigate the sporadic NANs in the stan analytical solutions (all dists) when the cdf is 1.

**Review by seabbs — COMMENTED** [review]
LGTM (self-review). Checked with consistency between stan and R + numerical vs analytical in R.

**Pull request merged**